clash-prelude/src/Clash/Explicit/BlockRam.hs

```diff
@@ -779,11 +779,11 @@ blockRam# clk content rd wen = case clockEnable clk of
           o'   = if re then ram V.! r else o
       in  o `seqX` o :- (ret `seq` rt `seq` et `seq` wt `seq` dt `seq` go' ram' o' res rs en wr din)
 
-    upd ram we waddr d = case maybeX we of
-      Nothing -> case maybeX waddr of
+    upd ram we waddr d = case maybeHasX we of
```

Comment:
I think `maybeIsX` is sufficient here

Reply:
You're right. `maybeHasX ~ maybeIsX` in this case, so I'm doing more work than needed!